tests/test_logic.py

```diff
@@ -1772,7 +1772,14 @@ def test_Controller_download_new_messages_with_new_message(mocker, session, sess
     failure_signal.connect.assert_called_once_with(
         co.on_message_download_failure, type=Qt.QueuedConnection
     )
-    set_status.assert_called_once_with("Retrieving new messages", 2500)
+
+    # fake APIJobQueue's emitting main_queue_updated when the job is queued and dequeued
+    co.api_job_queue.main_queue_updated.emit(1)
+    set_status.assert_called_once_with("Retrieving 1 message", 0)
+
```

Comment:
```suggestion

    # fake APIJobQueue's emitting main_queue_updated when the job is queued and dequeued
    co.api_job_queue.main_queue_updated.emit(1)
    set_status.assert_called_once_with("Retrieving 1 message", 0)
    
    co.api_job_queue.main_queue_updated.emit(2)
    set_status.assert_called_once_with("Retrieving 2 messages", 0)

```

Might be worth adding  one test case at the boundary between the singular and the plural forms of the message for completeness.

Reply:
Good call!